### PR TITLE
Fail on CI if tutorial notebooks raise errors

### PR DESCRIPTION
**Pull request by jsignell**
**Related Issue(s):** 

- #488 

**Description:**

Switch CI to use conda env in same as on read-the-docs, add missing conda-only dependencies to env, flip `nbsphinx_allow_errors` to False.

**PR Checklist:**

- [x] Code is formatted
- [x] Tests pass
- [ ] Changes are added to the [CHANGELOG](https://github.com/stac-utils/pystac-api-client/blob/main/CHANGELOG.md)

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -78,21 +78,22 @@ jobs:
         run: ./scripts/test
 
   docs:
+    if: matrix.python-version != "3.11"
```
Comment (jsignell) [inline .github/workflows/continuous-integration.yml:81]: the hvplot notebook fails on 3.11 I think there is some dependency issue that they need to fix upstream :(

Comment (gadomski) [reply .github/workflows/continuous-integration.yml:81]: Can you add a comment to that effect to remind future us of why we need this `if`?

**Review by gadomski — CHANGES_REQUESTED** [review]
Looks like the CI file isn't quite correct:

image

**Comment by jsignell** [comment]
I think I fixed the issue with the original workflow config, but now the pip section of the environment isn't resolving. There seem to be similar things going on with the other CI jobs that were untouched, so maybe this is coming from somewhere else....

**Comment by jsignell** [comment]
Ok! I made it out the other side on debugging the github action :sweat_smile:

**Review by gadomski — APPROVED** [review]

**Pull request merged**